optuna/_gp/_search_space.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import IntEnum
+import math
+from typing import Any
+
+import numpy as np
+import scipy.stats.qmc
+
+import optuna
+
+
+class ParamType(IntEnum):
```

Comment:
[nit] Probably, scale is more appropriate.
```suggestion
class ScaleType(IntEnum):
```